pallets/route-executor/src/lib.rs

```diff
@@ -268,18 +270,17 @@ pub mod pallet {
 			let who = ensure_signed(origin.clone())?;
 			Self::ensure_route_size(route.len())?;
 
-			let route = Self::get_route_or_default(route, AssetPair::new(asset_in, asset_out))?;
+			let asset_pair = AssetPair::new(asset_in, asset_out);
+			let route = Self::get_route_or_default(route, asset_pair)?;
+			Self::ensure_route_arguments(&asset_pair, &route)?;
 
 			let user_balance_of_asset_in_before_trade =
 				T::Currency::reducible_balance(asset_in, &who, Preservation::Preserve, Fortitude::Polite);
 
 			let trade_amounts = Self::calculate_buy_trade_amounts(&route, amount_out)?;
 
-			let last_trade_amount = trade_amounts.last().ok_or(Error::<T>::RouteCalculationFailed)?;
-			ensure!(
-				last_trade_amount.amount_in <= max_amount_in,
-				Error::<T>::TradingLimitReached
-			);
+			let first_trade = trade_amounts.last().ok_or(Error::<T>::RouteCalculationFailed)?;
```

Comment:
last() is first? please add test to ensure you have this right

Reply:
the `calculate_buy_trade_amounts` calculates the trades from the end as we have the `amount_out` specified. So the last of that list will be the `first_trade`, hence the renaming.

we cover this with test.

Reply:
but funcionally you didnt change anything, just renamed `last_trade_amount` to `first_trade`

Reply:
the problem is in the asset in vs route asset in inconsistency